src/connection/mod.rs

```diff
@@ -161,7 +171,11 @@ impl<C: FirebirdClient> Connection<C> {
 
     /// Run a closure with a transaction, if the closure returns an error
     /// and the default transaction is not active, the transaction will rollback, else it will be committed
-    pub fn with_transaction<T, F>(&mut self, closure: F) -> Result<T, FbError>
+    pub fn with_transaction<T, F>(
+        &mut self,
+        custom_confs: Option<TransactionConfiguration>,
+        closure: F,
+    ) -> Result<T, FbError>
```

Comment:
I would prefer the `with_transaction` to use the default transaction configuration and add a `with_transaction_config` or similar that you can pass the `TransactionConfiguration` option,
so the user does not need to pas a `None`or `Some`.

Reply:
Done. Looks better.